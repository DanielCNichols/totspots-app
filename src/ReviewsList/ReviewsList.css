.reviewslist {
  margin: 40px 0;
}

.reviewslist h3 {
  margin: 0;
}
