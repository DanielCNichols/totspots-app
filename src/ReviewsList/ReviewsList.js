import React, { Component } from 'react';
import venuesContext from '../VenuesContext';
import ApiService from '../services/api-service';
import UserReview from '../UserReview/UserReview';
import './ReviewsList.css';

export default class ReviewsList extends Component {
  static contextType = venuesContext;

  componentDidMount() {
    ApiService.getUserReviews()
      .then(reviews => {
        this.context.setUserReviews(reviews);
      })
      .catch(this.context.setError);
  }

  prerender() {
    let { userReviews } = this.context;
    if (userReviews.length === 0) {
      return (
        <div>
          <p>Looks like you haven't posted any reviews... yet!</p>
        </div>
      );
    } else {
      return (
        <section className="reviewslist">
          <header>
            <h3>Your Reviews</h3>
          </header>
          <div>
            <ul>
              {userReviews.map(userReviews => (
                <UserReview userReviews={userReviews} key={userReviews.id} />
              ))}
            </ul>
          </div>
        </section>
      );
    }
  }

  render() {
    return <div>{this.prerender()}</div>;
  }
}
