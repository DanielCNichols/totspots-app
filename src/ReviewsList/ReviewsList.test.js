import React from 'react';
import ReactDOM from 'react-dom'
import ReviewsList from './ReviewsList.js'
import {MemoryRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'



it('renders ReviewsList without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <MemoryRouter>
    <ReviewsList />
  </MemoryRouter>, div);
  ReactDOM.unmountComponentAtNode(div)
})

it('renders ReviewsList as expected', () => {
  const tree = renderer
  .create(
  <MemoryRouter>
    <ReviewsList name="ReviewsList"/>
  </MemoryRouter>
  )
  .toJSON();
  expect(tree).toMatchSnapshot();
})
