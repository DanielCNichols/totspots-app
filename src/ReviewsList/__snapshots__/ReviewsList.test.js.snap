// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ReviewsList as expected 1`] = `
<section
  className="reviewslist"
>
  <header>
    <h3>
      Your Reviews
    </h3>
  </header>
  <div>
    <ul />
  </div>
</section>
`;
